Update ASI04_Agentic_Supply_Chain_Vulnerabilities .md

syedDS opened the pull request:
pushing the Gdoc draft in for review and comments

syedDS committed:
Update ASI04_Agentic_Supply_Chain_Vulnerabilities .md

pushing the Gdoc draft in for review

Signed-off-by: syedDS <32578528+syedDS@users.noreply.github.com>

itskerenkatz commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
"leading it to execute unintended risky behavior"
Harmful usually aligns with hate, violence speech and I guess you meant it in a broader way, right?

syedDS replied:
Added

itskerenkatz commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
What about: Symbol attack - when tools or agents are faking to be tools or agents that they are not

itskerenkatz commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
Maybe I haven’t understood correctly, but if it’s in a sandbox, how will it communicate with external tools or agents?

syedDS replied:
Updated context to better showcase what does sandbox mean here!

itskerenkatz commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
This one is more ASI03 and feels to me a bit redundant here

syedDS replied:
Updated

itskerenkatz commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
This one is ASI06

syedDS replied:
Ok actually these controls mitigate downstream ASI06 memory and context poisoning risks by enforcing agentic supply-chain integrity at ingestion and runtime.

itskerenkatz commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
**against** poisoned components

itskerenkatz commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
I think it's better to name it as: "Hallucinated resource" to be more generic and focus on the risk rather than the incident

syedDS replied:
updated

itskerenkatz commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
A bit more related to ASI05, what do you think?

syedDS replied:
Updated

itskerenkatz reviewed (changes requested):
Fantastic work! Added a few comments

syedDS committed:
Merge branch 'main' into patch-1

syedDS committed:
Update ASI04_Agentic_Supply_Chain_Vulnerabilities .md

- Redefined Agentic Supply Chain definition 
- Incorporated new Attacks 
- Incorporated the review comments

Signed-off-by: syedDS <32578528+syedDS@users.noreply.github.com>

syedDS commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
Added

syedDS committed:
Update ASI04_Agentic_Supply_Chain_Vulnerabilities .md

Added context to  attack where malicious service deliberately impersonates a legitimate tool or agent

Signed-off-by: syedDS <32578528+syedDS@users.noreply.github.com>

omerhof-fujitsu commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
Minor: Consider removing "in two ways" for shortening the sentence a bit.

omerhof-fujitsu commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
Minor: consider replace with: "unless fully verified against"

omerHof commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
I noticed that the document differentiates ASI04 from other risk categories (LLM03:2025, ASI02, ASI07, and traditional AppSec supply-chain risks) in three different sections of the text. While the explanations are correct, spreading them across the document makes it harder for readers to understand the scope and taxonomy relationships.
To improve clarity and maintainability, I suggest consolidating all these comparisons into a single dedicated section, such as “Scope & Differentiation from Related Risks.” This section could briefly compare ASI04 against LLM03 (static LLM supply chain), ASI02 (Tool Misuse), ASI07 (Rogue/Impersonated Agents), and traditional AppSec supply-chain risks. This would help readers quickly grasp how ASI04 fits into the broader OWASP GenAI and Agentic Security taxonomy, without repeating similar explanations in multiple places.

syedDS replied:
Good suggestion , we planned to do separate section in V2, as this was kept brief only in scope of  ASI04

omerHof commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
11. [https://embracethered.com/blog/posts/2025/cross-agent-privilege-escalation-agents-that-free-each-other/](https://embracethered.com/blog/posts/2025/cross-agent-privilege-escalation-agents-that-free-each-other/)

syedDS replied:
updated

omerHof reviewed (commented):
I’ve completed my review, the content is written clearly and interesting!
I added a few comments, nothing major.

syedDS commented on initiatives/agent_security_initiative/agentic-top-10/Sprint 1-first-public-draft-expanded/ASI04_Agentic_Supply_Chain_Vulnerabilities .md:
This attack is more aligned to ASI04 than replit so replacing

syedDS reviewed (commented):
Resolved all the comment

syedDS reviewed (commented):
resolved

syedDS committed:
Merge branch 'main' into patch-1

syedDS commented:
opened a new PR so closing this one :review and resolved comments ASI04: supply chain vulnerabilities #793
